soh/soh/Enhancements/randomizer/randomizer_item_tracker.cpp

```diff
@@ -659,16 +664,20 @@ void DrawDungeonItem(ItemTrackerItem item) {
                      ImVec2(iconSize, iconSize), ImVec2(0, 0), ImVec2(1, 1));
     }
 
-    if (ResourceMgr_IsSceneMasterQuest(item.data) && (CHECK_DUNGEON_ITEM(DUNGEON_MAP, item.data) || item.data == SCENE_GERUDO_TRAINING_GROUND || item.data == SCENE_INSIDE_GANONS_CASTLE)) {
-        dungeonColor = IM_COL_PURPLE;
+    if (CheckTracker::IsAreaSpoiled(RandomizerCheckObjects::GetRCAreaBySceneID((SceneID)item.data)) && GameInteractor::IsSaveLoaded()) {
+        if (ResourceMgr_IsSceneMasterQuest(item.data))
+            dungeonColor = IM_COL_PURPLE;
+        else
+            dungeonColor = IM_COL_LIGHT_YELLOW;
```

Comment:
if statements without braces scare me, but in this case it can be simplified to

```suggestion
        dungeonColor = ResourceMgr_IsSceneMasterQuest(item.data) ? IM_COL_PURPLE : IM_COL_LIGHT_YELLOW;
```